sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -547,7 +547,17 @@ case class Cast(child: Expression, dataType: DataType, timeZoneId: Option[String
   private[this] def castToDouble(from: DataType): Any => Any = from match {
     case StringType =>
       buildCast[UTF8String](_, s => try s.toString.toDouble catch {
-        case _: NumberFormatException => null
+        case _: NumberFormatException =>
+          val str = s.trim.toString
```

Comment:
Its just a check; do we support this case, e.g., '   infinity   '? (I know pg does so though). If we support this case, IMO you'd be better to clearly describe in the PR description, which case this PR try to support.

Reply:
@maropu Currently i trim spaces from both ends and compare. So this case should work. 
Ok.. i will update the description.